rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1661,6 +1661,87 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
 		and packageRef count = openTradeNoExecutionDetails count
 		and packageRef distinct count = 1
 
+
+func Create_PartialDeliveryPrimitiveInstruction: <"Creates the primitive instructions for a substitution of collateral by replacing the assetpayout of the trade.">
+	inputs:
+		//ICMA-P2 Instructions
```

Comment:
What are ICMA-P2 Instructions? Is this an ICMA specific function?

Reply:
A partial delivery occurs when the collateral giver is unable to deliver the full amount on settlement. The market best practice recommends buyers accept the amount available for delivery and may negotiate the remaining. So the workflow in the function is to modify the existing contract to accept the available delivered collateral with an option to open a new contract on the undelivered amount.